### ISSUE TITLE
questions about OKNet code and model training

### ISSUE THREAD

**Issue by TZhaoPKU** [c-yn/OKNet#8]
Hey authors. I have two questions about your code and model training:)

Q1: I noticed that the code of OKM model is not consistent with the description in your paper: 
According to the paper (Fig.2(b)), the shortcut path is combined with other branches **after** 1x1 conv 
However, in the code (OKNet.py line 90), the shortcut path (x) is combined **before** 1x1 conv:
_out = x + self.dw_13(out) + self.dw_31(out) + self.dw_33(out) + self.dw_11(out) + x_sca_

Would you please let me know which way would be recommended based on your experiences? 

Q2: I am trying to train the OKNet model and found that training cost had sharp impulses (like jumping to ~10^3 from ~10^-2 and drop back again) from time to time. I'm pasting an example here:
![Image](https://github.com/user-attachments/assets/77c87518-8ea3-42b4-8c27-4778c3ff0b25)

Just wondering did you also notice similar phenomenon in your training? If yes, any potential reasons? 

Thank you in advance!

Tingting

**Comment by c-yn** [comment]
Hi Tingting,

1, Thanks for your interest. Yes, there is an inconsistency between the paper and the code regarding the starting point of the shortcut path. The code version produced the results reported in the paper. I haven't investigated which version was better.

2, The sharp impulses may be due to a large initial learning rate. If the loss keeps decreasing, I think you can continue training. To be honest, I don't remember if I encountered this issue.

**Comment by TZhaoPKU** [comment]
Got it. Thank you. Really appreciate your quick response and advices!!

Tingting

**Comment by c-yn** [comment]
you are welcome
